bulbs/reading_list/mixins.py

```diff
@@ -0,0 +1,186 @@
+"""Logic for reading list behavior across all properties."""
+from django.conf import settings
+from django.core.cache import cache
+
+from elasticsearch import TransportError
+from elasticsearch_dsl import filter as es_filter
+
+from bulbs.content.filters import NegateQueryFilter, SponsoredBoost
+from bulbs.content.models import Content
+from bulbs.content.search import randomize_es
+from bulbs.sections.models import Section
+from bulbs.special_coverage.models import SpecialCoverage
+from .popular import get_popular_ids, popular_content
+from .slicers import FirstSlotSlicer
+
+
+READING_LIST_CONFIG = getattr(settings, "READING_LIST_CONFIG", {})
+
+
+class ReadingListMixin(object):
+    """Mixin for Content-based objects to manage reading lists."""
+
+    def _get_reading_list_identifier(self):
+
+        # 1. Match content to sponsored Special Coverages
+        results = self.percolate_special_coverage(sponsored_only=True)
+        if results:
+            return results[0]
+
+        # 2."Popular" i.e., the content is one of the 25 most popular items.
+        popular_ids = get_popular_ids()
+        if popular_ids and self.id in popular_ids:
+            return "popular"
+
+        # 3. Any unsponsored special coverage reading list that contains this item.
+        results = self.percolate_special_coverage()
+        if results:
+            return results[0]
+
+        # 4. Any section that contains this item.
+        try:
+            results = Content.search_objects.client.percolate(
```

Comment:
should be refactored into a Content.percolate_sections() method, similar to special coverage